src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -22,7 +31,18 @@ const mapExternalLanguageToOptions = (language: string) => {
   };
 };
 
-export const QueryLanguageSelector = (props: Props) => {
+export const QueryLanguageSelector = (props: QueryLanguageSelectorProps) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
```

Comment:
nit: do we need these? these are just one line anyways

Reply:
agreed, made it one liner to reduce code